### PR TITLE
Irrlicht cleanup: cleanup various object to use RenderingEngine

### PR DESCRIPTION
**Pull request by nerzhul**
* CAO doesn't need scenemanager in addToScene
* Camera doesn't need VideoDriver pointer or SceneManager in constructor
* Hud doesn't need driver & scene manager in constructor
* Hud doesn't need scenemanager pointer
* Tile.h doesn't need IrrlichtDevice header (just SMaterial)

### CODE REVIEW

File: src/hud.cpp [hunk 1]
```diff
@@ -40,12 +40,10 @@ with this program; if not, write to the Free Software Foundation, Inc.,
...
 
-Hud::Hud(video::IVideoDriver *driver, scene::ISceneManager* smgr,
-		gui::IGUIEnvironment* guienv, Client *client, LocalPlayer *player,
+Hud::Hud(gui::IGUIEnvironment* guienv, Client *client, LocalPlayer *player,
 		Inventory *inventory)
 {
-	this->driver      = driver;
-	this->smgr        = smgr;
+	driver      = RenderingEngine::get_video_driver();
 	this->guienv      = guienv;
```
Comment (SmallJoker) [inline src/hud.cpp]: `=` (of `driver`) should be aligned the same way as these assignments.

File: src/hud.cpp [hunk 2]
```diff
@@ -40,12 +40,10 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "touchscreengui.h"
 #endif
 
-Hud::Hud(video::IVideoDriver *driver, scene::ISceneManager* smgr,
-		gui::IGUIEnvironment* guienv, Client *client, LocalPlayer *player,
+Hud::Hud(gui::IGUIEnvironment* guienv, Client *client, LocalPlayer *player,
```
Comment (SmallJoker) [inline src/hud.cpp]: `gui::IGUIEnvironment *guienv`

File: src/content_cao.cpp [hunk 3]
```diff
@@ -865,8 +865,9 @@ void GenericCAO::addToScene(scene::ISceneManager *smgr, ITextureSource *tsrc)
 			infostream << "serialized form: " << m_prop.wield_item << std::endl;
 			item.deSerialize(m_prop.wield_item, m_client->idef());
 		}
-		m_wield_meshnode = new WieldMeshSceneNode(smgr->getRootSceneNode(),
-			smgr, -1);
+		m_wield_meshnode = new WieldMeshSceneNode(
+			RenderingEngine::get_scene_manager()->getRootSceneNode(),
+			RenderingEngine::get_scene_manager(), -1);
```
Comment (SmallJoker) [inline src/content_cao.cpp]: I guess these arguments (class `WieldMeshSceneNode`) will be cleaned up later?

File: src/camera.h [hunk 4]
```diff
@@ -60,8 +60,7 @@ enum CameraMode {CAMERA_MODE_FIRST, CAMERA_MODE_THIRD, CAMERA_MODE_THIRD_FRONT};
 class Camera
 {
 public:
-	Camera(scene::ISceneManager* smgr, MapDrawControl& draw_control,
-			Client *client);
+	Camera(MapDrawControl& draw_control, Client *client);
```
Comment (SmallJoker) [inline src/camera.h]: `MapDrawControl &draw_control`

File: src/camera.cpp [hunk 5]
```diff
@@ -38,16 +38,11 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #define CAMERA_OFFSET_STEP 200
 
-#include "nodedef.h"
-
-Camera::Camera(scene::ISceneManager* smgr, MapDrawControl& draw_control,
-		Client *client):
+Camera::Camera(MapDrawControl& draw_control, Client *client):
```
Comment (SmallJoker) [inline src/camera.cpp]: `MapDrawControl &draw_control` here as well

File: src/hud.h [hunk 6]
```diff
@@ -120,8 +120,7 @@ class Hud {
 	std::string hotbar_selected_image = "";
 	bool use_hotbar_selected_image = false;
 
-	Hud(video::IVideoDriver *driver,scene::ISceneManager* smgr,
-		gui::IGUIEnvironment* guienv, Client *client, LocalPlayer *player,
+	Hud(gui::IGUIEnvironment* guienv, Client *client, LocalPlayer *player,
```
Comment (SmallJoker) [inline src/hud.h]: Pointer `guienv` again

**Comment by nerzhul** [comment]
i missed the old style, will fix it this evening :)

File: src/content_cao.cpp (hunk 3 above)
Comment (nerzhul) [inline src/content_cao.cpp]: yes i think i will do it later, it was a first analysis about scenemgr but no time atm to continue this PR, waiting for merge to continue

Comment (nerzhul) [inline src/content_cao.cpp]: in fact it was easy to fix, i did it

**Pull request merged**